sdk/python/kfp/components/pipeline_task.py

```diff
@@ -232,13 +228,14 @@ def expand_command_part(arg) -> Union[str, List[str], None]:
                 if input_name in args or type_utils.is_task_final_status_type(
                         inputs_dict[input_name].type):
                     return arg.to_placeholder_string()
+
+                input_spec = inputs_dict[input_name]
+                if input_spec.default is None:
```

Comment:
I might miss some context here. Isn't default value always optional at compile time?

Reply:
Synced offline, this logic--requiring default--doesn't seem to be right, but it exists before this PR. We can go ahead merge this PR first and investigate this logic separately.